frontend/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,13 @@
+import {Navigate, Outlet} from "react-router-dom";
+
+type ProtectedRouteProps = {
+    user: string | undefined;
+}
+
+export default function ProtectedRoute(props: ProtectedRouteProps){
+    const isAuthenicated = props.user !== "anonymousUser" && props.user !== null;
```

Comment:
props.user can never be null since your type only allowed string or undefined (Line 3-5).